### ISSUE TITLE
Add __repr__ methods

### ISSUE THREAD

**Issue by blueyed** [Delgan/loguru#84]
I think it would be helpful to have `__repr__` methods in general, but especially for `Logger` and `Handler`.

What's your opinion on this?
Which style do you prefer?

**Comment by Delgan** [comment]
What would be your use case? I don't remember ever having to call `print(logger)`. 😄 
It makes sense for the standard `logging` logger to have a custom representation, because each built-in logger is associated with a name and a level. But the Loguru logger is just a dumb object, there is not much to display. 

`Handler` is not even supposed to be part of the public API! I don't see reason to add a `__repr__` to it.

**Comment by blueyed** [comment]
My use case is looking at in the debugger.

Logger could include the number of handlers it has for example (`_handlers_count`), or just the `repr()` of `_handlers`, where Handler's repr would then include `_writer` and `_levelno`.

**Comment by Delgan** [comment]
Fine, if that might be useful, a custom representation with the number of handlers seems good.

Something like `<loguru.logger (4 handlers)>`?

**Comment by blueyed** [comment]
Or `<loguru.logger handlers=%r>` already, where `%r` would expand to something like `{0: <loguru.handler level=4 writer=%r>, …}` then, i.e. it would provide info about the handlers themselves already.
Not sure how to represent writer though, using its repr might become too long, but I guess its ok.

This is only meant to show up in debuggers, tracebacks etc, where more information is usually better.

`<loguru.logger (4 handlers)>` could be used for `__str__` maybe.

**Comment by Delgan** [comment]
I made a try with the current implementation and two handlers, it looks it would require some refactoring:

```
<loguru.logger handlers={0: <loguru.handler level=10 writer=<function Logger.add.<locals>.writer at 0x000001399785EC80>>, 1: <loguru.handler level=100 writer=<bound method FileSink.write of <loguru._file_sink.FileSink object at 0x00000139979BE208>>>}>
```

Actually, I'm a little bit worried about exposing internals information through the `logger` representation... Like, `Handler` objects are not even _supposed_ to exist. Also, there is no way to get a list of the added handlers and to know what are their identifiers, except by... Printing the `logger`? Is it ok? 😕

**Comment by Delgan** [comment]
I finally implemented this as you suggested.

My concerns about exposing internals made little sense. The benefits and convenience of ``__repr__`` for debugging outweighs the purity violation that I feared by "leaking" implementation details. Printing the ``logger`` can be used for debugging and that's all.

I tried to keep the representation quite "compact", so it will look like `"<loguru.logger handlers={0: (level=10, writer=<stderr>)}>"`. Depending on the `writer`'s type, its representation is retrieved with `.name`, `.__name__` or `repr()`.
